server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -575,97 +496,50 @@ private ReserveCompactionTask(CompactionJobQueues.MetaJob metaJob, String compac
-            .incrementBy(Duration.ofMillis(100)).maxWait(Duration.ofSeconds(10)).backOffFactor(1.5)
-            .logInterval(Duration.ofMinutes(3)).createRetry();
-
-        while (retry.canRetry()) {
-          try (var tabletsMutator = ctx.getAmple().conditionallyMutateTablets()) {
-            var extent = metaJob.getTabletMetadata().getExtent();
-
-            if (!canReserveCompaction(tabletMetadata, metaJob.getJob().getKind(), jobFiles, ctx,
-                manager.getSteadyTime())) {
-              return null;
-            }
-
-            var ecm = createExternalCompactionMetadata(metaJob.getJob(), jobFiles, tabletMetadata,
-                compactorAddress, externalCompactionId);
-
-            // any data that is read from the tablet to make a decision about if it can compact or
-            // not
-            // must be checked for changes in the conditional mutation.
-            var tabletMutator = tabletsMutator.mutateTablet(extent).requireAbsentOperation()
-                .requireFiles(jobFiles).requireNotCompacting(jobFiles);
-            if (metaJob.getJob().getKind() == CompactionKind.SYSTEM) {
-              // For system compactions the user compaction requested column is examined when
-              // deciding
-              // if a compaction can start so need to check for changes to this column.
-              tabletMutator.requireSame(tabletMetadata, SELECTED, USER_COMPACTION_REQUESTED);
-            } else {
-              tabletMutator.requireSame(tabletMetadata, SELECTED);
-            }
-
-            if (metaJob.getJob().getKind() == CompactionKind.SYSTEM) {
-              var selectedFiles = tabletMetadata.getSelectedFiles();
-              var reserved =
-                  getFilesReservedBySelection(tabletMetadata, manager.getSteadyTime(), ctx);
-
-              // If there is a selectedFiles column, and the reserved set is empty this means that
-              // either no user jobs were completed yet or the selection expiration time has passed
-              // so the column is eligible to be deleted so a system job can run instead
-              if (selectedFiles != null && reserved.isEmpty()
-                  && !Collections.disjoint(jobFiles, selectedFiles.getFiles())) {
-                LOG.debug("Deleting user compaction selected files for {} {}", extent,
-                    externalCompactionId);
-                tabletMutator.deleteSelectedFiles();
-              }
-            }
+      if (ctx.getTableState(rcJob.getExtent().tableId()) != TableState.ONLINE) {
+        return null;
+      }
 
-            tabletMutator.putExternalCompaction(externalCompactionId, ecm);
-            tabletMutator.submit(
-                tm -> tm.getExternalCompactions().containsKey(externalCompactionId),
-                () -> "compaction reservation");
+      try {
+        try (var tabletsMutator = ctx.getAmple().conditionallyMutateTablets()) {
```

Comment:
the changes here are really nice, it's definitely a lot simpler. I think it would make moving the reservation to the Compactor easier as well if we implement #4978